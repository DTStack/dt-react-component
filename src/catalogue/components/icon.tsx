import React from 'react';
import classNames from 'classnames';

interface IIcon {
    className?: string;
    style?: React.CSSProperties;
    onClick?: React.MouseEventHandler<HTMLSpanElement>;
}

const FolderIcon = function ({ className, ...rest }: IIcon) {
    return (
        <span className={classNames('dt-catalogue__icon', className)} {...rest}>
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 64 64"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
            >
                <g>
                    <g>
                        <path
                            fillRule="evenodd"
                            clipRule="evenodd"
                            d="M32.1355 19.1579H52.4464C53.8565 19.1595 54.9991 20.3073 55.0007 21.7237V25.0281H8.00073V15.5658C8.00232 14.1494 9.14501 13.0016 10.5551 13H24.7355C25.4134 12.9981 26.0638 13.2687 26.5418 13.7515L31.7744 19.0075C31.8698 19.1042 31.9999 19.1583 32.1355 19.1579ZM8.00073 28.0281V49.4342C8.00232 50.8506 9.14501 51.9984 10.5551 52H52.4464C53.8565 51.9984 54.9991 50.8506 55.0007 49.4342V28.0281H8.00073Z"
                            fill="currentColor"
                        />
                    </g>
                </g>
            </svg>
        </span>
    );
};

const FolderOpenedIcon = function ({ className, ...rest }: IIcon) {
    return (
        <span className={classNames('dt-catalogue__icon', className)} {...rest}>
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 64 64"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
            >
                <g>
                    <g>
                        <path
                            fillRule="evenodd"
                            clipRule="evenodd"
                            d="M11.9214 28.719L8 43.0351V15.5658C8.0016 14.1494 9.15438 13.0016 10.5769 13H22.8211C23.5049 12.9981 24.1611 13.2687 24.6433 13.7515L29.9221 19.0075C30.0184 19.1042 30.1496 19.1583 30.2864 19.1579H48.7151C50.1377 19.1595 51.2905 20.3073 51.2921 21.7237V25.3158H16.3962C14.3057 25.3192 12.4747 26.7117 11.9214 28.719ZM56.4697 28.3768C55.9841 27.7389 55.2256 27.3654 54.4217 27.3684H16.3962C15.2344 27.3706 14.217 28.145 13.9101 29.2607L8.56885 48.7607C8.35812 49.532 8.52044 50.357 9.00787 50.9921C9.49529 51.6272 10.2521 51.9999 11.0549 52H49.0804C50.2422 51.9978 51.2596 51.2235 51.5665 50.1077L56.9078 30.6077C57.1215 29.8368 56.9591 29.0107 56.4694 28.3768H56.4697Z"
                            fill="currentColor"
                        />
                    </g>
                </g>
            </svg>
        </span>
    );
};

const DownTriangleIcon = function ({ className, ...rest }: IIcon) {
    return (
        <span className={classNames('dt-catalogue__icon', className)} {...rest}>
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 64 64"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <g>
                    <path
                        fillRule="evenodd"
                        clipRule="evenodd"
                        d="M47.791 27.4819L33.0613 42.4125C32.3091 43.175 31.0812 43.1833 30.3187 42.431L15.8855 27.4661C15.1419 26.6951 15.1642 25.4674 15.9351 24.7238C16.2966 24.3752 16.7792 24.1804 17.2814 24.1804L46.4104 24.1804C47.4815 24.1804 48.3498 25.0487 48.3498 26.1198C48.3498 26.6296 48.1491 27.1189 47.791 27.4819Z"
                        fill="currentColor"
                    />
                </g>
            </svg>
        </span>
    );
};

const CatalogIcon = function ({ className, ...rest }: IIcon) {
    return (
        <span className={classNames('dt-catalogue__icon', className)} {...rest}>
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 64 64"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <g>
                    <path
                        fillRule="evenodd"
                        clipRule="evenodd"
                        d="M52 8C54.2091 8 56 9.79086 56 12V52C56 54.2091 54.2091 56 52 56H12C9.79086 56 8 54.2091 8 52V12C8 9.79086 9.79086 8 12 8H52ZM45.7468 39.6385H28.0755C26.9741 39.6385 26.078 40.5355 26.078 41.6382C26.078 42.6907 26.8945 43.5558 27.9267 43.6323L28.0755 43.6378H45.7468C46.847 43.6378 47.7422 42.7408 47.7422 41.6382C47.7422 40.5857 46.9266 39.7205 45.8955 39.644L45.7468 39.6385ZM21.7673 39.6385H18.519C17.4187 39.6385 16.5235 40.5355 16.5235 41.6382C16.5235 42.6907 17.3391 43.5558 18.3703 43.6323L18.519 43.6378H21.7673C22.8688 43.6378 23.7649 42.7408 23.7649 41.6382C23.7649 40.5857 22.9484 39.7205 21.9162 39.644L21.7673 39.6385ZM45.7468 29.708H28.0755C26.9741 29.708 26.078 30.605 26.078 31.7076C26.078 32.7601 26.8945 33.6253 27.9267 33.7018L28.0755 33.7073H45.7468C46.847 33.7073 47.7422 32.8102 47.7422 31.7076C47.7422 30.6552 46.9266 29.79 45.8955 29.7135L45.7468 29.708ZM21.7673 29.708H18.519C17.4187 29.708 16.5235 30.605 16.5235 31.7076C16.5235 32.7601 17.3391 33.6253 18.3703 33.7018L18.519 33.7073H21.7673C22.8688 33.7073 23.7649 32.8102 23.7649 31.7076C23.7649 30.6552 22.9484 29.79 21.9162 29.7135L21.7673 29.708ZM45.7468 19.7774H28.0755C26.9741 19.7774 26.078 20.6744 26.078 21.7771C26.078 22.8296 26.8945 23.6948 27.9267 23.7712L28.0755 23.7767H45.7468C46.847 23.7767 47.7422 22.8797 47.7422 21.7771C47.7422 20.7246 46.9266 19.8594 45.8955 19.7829L45.7468 19.7774ZM21.7673 19.7774H18.519C17.4187 19.7774 16.5235 20.6744 16.5235 21.7771C16.5235 22.8296 17.3391 23.6948 18.3703 23.7712L18.519 23.7767H21.7673C22.8688 23.7767 23.7649 22.8797 23.7649 21.7771C23.7649 20.7246 22.9484 19.8594 21.9162 19.7829L21.7673 19.7774Z"
                        fill="currentColor"
                    />
                </g>
            </svg>
        </span>
    );
};

const DragIcon = function ({ className, ...rest }: IIcon) {
    return (
        <span className={classNames('dt-catalogue__icon', className)} {...rest}>
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 64 64"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <g>
                    <path
                        fillRule="evenodd"
                        clipRule="evenodd"
                        d="M30.7954 5.61121C31.5562 4.82287 32.7988 4.79924 33.5889 5.52808L33.7097 5.64941L40.4525 13.0137C41.1984 13.8284 41.1427 15.0935 40.328 15.8394C39.5562 16.5461 38.3801 16.5332 37.6243 15.8372L37.5023 15.7149L34.2351 12.1471V30.5029H52.7644L49.4815 27.3357C48.7285 26.6091 48.6686 25.4345 49.3166 24.637L49.4311 24.5078C50.1578 23.7548 51.3324 23.6949 52.1298 24.3428L52.2591 24.4574L59.6234 31.5638C60.4118 32.3245 60.4354 33.5671 59.7066 34.3572L59.5852 34.478L52.2209 41.2208C51.4062 41.9667 50.1411 41.911 49.3952 41.0963C48.6886 40.3245 48.7014 39.1485 49.3974 38.3926L49.5197 38.2706L53.635 34.5029H34.2351V51.2187L37.5028 47.6494L37.6247 47.5271C38.3806 46.8311 39.5567 46.8183 40.3284 47.5249C41.1431 48.2708 41.1989 49.5359 40.4529 50.3506L33.7102 57.7149L33.5893 57.8363C32.7993 58.5651 31.5566 58.5415 30.7959 57.7531L23.6895 50.3888L23.5749 50.2595C22.927 49.4621 22.9869 48.2875 23.7399 47.5608L23.8692 47.4463C24.6666 46.7983 25.8412 46.8582 26.5679 47.6112L30.2351 51.4122V34.5029H10.6026L14.7149 38.2677L14.8372 38.3896C15.5332 39.1455 15.5461 40.3216 14.8394 41.0934C14.0935 41.908 12.8284 41.9638 12.0137 41.2179L4.64941 34.4751L4.52808 34.3542C3.79924 33.5642 3.82287 32.3215 4.61121 31.5608L11.9755 24.4544L12.1048 24.3399C12.9023 23.6919 14.0769 23.7518 14.8035 24.5048L14.9181 24.6341C15.566 25.4316 15.5061 26.6062 14.7531 27.3328L11.4674 30.5029H30.2351V11.9519L26.5674 15.7531C25.8408 16.5061 24.6662 16.566 23.8687 15.9181L23.7394 15.8035C22.9864 15.0769 22.9265 13.9023 23.5745 13.1048L23.689 12.9755L30.7954 5.61121Z"
                    />
                </g>
            </svg>
        </span>
    );
};

const EllipsisIcon = function ({ className, ...rest }: IIcon) {
    return (
        <span className={classNames('dt-catalogue__icon', className)} {...rest}>
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 64 64"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <g>
                    <path
                        fillRule="evenodd"
                        clipRule="evenodd"
                        d="M47 32C47 34.7614 49.2386 37 52 37C54.7614 37 57 34.7614 57 32C57 29.2386 54.7614 27 52 27C49.2386 27 47 29.2386 47 32ZM27 32C27 34.7614 29.2386 37 32 37C34.7614 37 37 34.7614 37 32C37 29.2386 34.7614 27 32 27C29.2386 27 27 29.2386 27 32ZM7 32C7 34.7614 9.23858 37 12 37C14.7614 37 17 34.7614 17 32C17 29.2386 14.7614 27 12 27C9.23858 27 7 29.2386 7 32Z"
                    />
                </g>
            </svg>
        </span>
    );
};

const SearchIcon = function ({ className, ...rest }: IIcon) {
    return (
        <span className={classNames('dt-catalogue__icon', className)} {...rest}>
            <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                viewBox="0 0 64 64"
                fill="currentColor"
            >
                <path
                    fillRule="evenodd"
                    clipRule="evenodd"
                    d="M30.1927 5C43.6716 5 54.3855 15.9572 54.3855 29.7421C54.3855 43.527 43.6716 54.4842 30.1927 54.4842C16.7139 54.4842 6 43.527 6 29.7421C6 15.9572 16.7139 5 30.1927 5ZM30.1927 9.52055C19.2345 9.52055 10.5217 18.4311 10.5217 29.7421C10.5217 41.053 19.2345 49.9636 30.1927 49.9636C41.151 49.9636 49.8638 41.053 49.8638 29.7421C49.8638 18.4311 41.151 9.52055 30.1927 9.52055ZM51.165 49.982C50.2767 49.2338 48.9479 49.2778 48.1115 50.1141C47.2285 50.9968 47.2285 52.4279 48.1115 53.3106L54.1405 59.338L54.2843 59.47C55.1725 60.2182 56.5014 60.1742 57.3378 59.338C58.2207 58.4553 58.2207 57.0242 57.3378 56.1415L51.3088 50.1141L51.165 49.982Z"
                    fill="currentColor"
                />
            </svg>
        </span>
    );
};

const CloseIcon = function ({ className, ...rest }: IIcon) {
    return (
        <span className={classNames('dt-catalogue__icon', className)} {...rest}>
            <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                viewBox="0 0 64 64"
                fill="currentColor"
            >
                <path
                    fillRule="evenodd"
                    clipRule="evenodd"
                    d="M51.0918 12.9081C51.8729 13.6892 51.8729 14.9555 51.0918 15.7365L34.8284 32L51.0918 48.2635C51.8729 49.0445 51.8729 50.3108 51.0918 51.0919C50.3108 51.8729 49.0444 51.8729 48.2634 51.0919L31.9999 34.8284L15.7365 51.0919C14.9554 51.8729 13.6891 51.8729 12.9081 51.0919C12.127 50.3108 12.127 49.0445 12.9081 48.2635L29.1715 32L12.9081 15.7365C12.127 14.9555 12.127 13.6892 12.9081 12.9081C13.6891 12.1271 14.9554 12.1271 15.7365 12.9081L31.9999 29.1716L48.2634 12.9081C49.0444 12.1271 50.3108 12.1271 51.0918 12.9081Z"
                    fill="currentColor"
                />
            </svg>
        </span>
    );
};

export {
    CatalogIcon,
    CloseIcon,
    DownTriangleIcon,
    DragIcon,
    EllipsisIcon,
    FolderIcon,
    FolderOpenedIcon,
    SearchIcon,
};
