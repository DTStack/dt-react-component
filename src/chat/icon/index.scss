.dtc__icon {
    display: inline-flex;
    align-items: center;
    color: inherit;
    font-style: normal;
    line-height: 0;
    text-align: center;
    text-transform: none;
    vertical-align: -0.125em;
    text-rendering: optimizelegibility;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    > * {
        line-height: 1;
    }
    svg {
        display: inline-block;
    }
}
