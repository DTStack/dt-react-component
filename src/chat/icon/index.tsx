import React from 'react';
import classNames from 'classnames';

import './index.scss';

export interface IconProps extends React.HTMLAttributes<HTMLSpanElement> {}

export function SendColored({ className, ...rest }: IconProps) {
    return (
        <span className={classNames('dtc__icon', className)} {...rest}>
            <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                fill="none"
                viewBox="0 0 24 24"
            >
                <path
                    fill={'url(#paint0_linear_3878_6538)'}
                    fillRule="evenodd"
                    d="M17.999 3.858 3.35 9.204a1.66 1.66 0 0 0-1.095 1.458 1.66 1.66 0 0 0 .862 1.602l4.454 2.44c.27.143.593.098.808-.118l.737-.72 4.508-4.428a.677.677 0 0 1 .952.018c.26.261.26.693-.009.954l-4.562 4.473-.61.603a.68.68 0 0 0-.118.828l2.64 4.626c.315.513.845.81 1.429.81.063 0 .117 0 .18-.009a1.68 1.68 0 0 0 1.41-1.107L20.153 6a1.69 1.69 0 0 0-.404-1.764 1.64 1.64 0 0 0-1.751-.378"
                    clipRule="evenodd"
                ></path>
                <defs>
                    <linearGradient
                        id="paint0_linear_3878_6538"
                        x1="2.25"
                        x2="23.921"
                        y1="3.75"
                        y2="11.552"
                        gradientUnits="userSpaceOnUse"
                    >
                        <stop stopColor="#00BAC6"></stop>
                        <stop offset="0.5" stopColor="#0067FF"></stop>
                        <stop offset="1" stopColor="#450FDE"></stop>
                    </linearGradient>
                </defs>
                <defs>
                    <linearGradient
                        id="paint0_linear_3878_6538_hover"
                        x1="2.25"
                        x2="23.921"
                        y1="3.75"
                        y2="11.552"
                        gradientUnits="userSpaceOnUse"
                    >
                        <stop stopColor="#08C4FF"></stop>
                        <stop offset="0.5" stopColor="#4892FF"></stop>
                        <stop offset="1" stopColor="#8A61FF"></stop>
                    </linearGradient>
                </defs>
            </svg>
        </span>
    );
}

export const GradientDotIcon = ({ className, ...rest }: IconProps) => {
    return (
        <span className={classNames('dtc__icon', className)} {...rest}>
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 6 6"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path
                    d="M0 3C0 1.34315 1.34315 0 3 0C4.65685 0 6 1.34315 6 3C6 4.65685 4.65685 6 3 6C1.34315 6 0 4.65685 0 3Z"
                    fill="url(#paint0_linear_5253_12831)"
                />
                <circle cx="3" cy="3" r="3" fill="url(#paint1_linear_5253_12831)" />
                <defs>
                    <linearGradient
                        id="paint0_linear_5253_12831"
                        x1="0"
                        y1="0"
                        x2="7.2238"
                        y2="2.60057"
                        gradientUnits="userSpaceOnUse"
                    >
                        <stop stopColor="#08C4FF" />
                        <stop offset="0.5" stopColor="#4892FF" />
                        <stop offset="1" stopColor="#8A61FF" />
                    </linearGradient>
                    <linearGradient
                        id="paint1_linear_5253_12831"
                        x1="0"
                        y1="0"
                        x2="7.2238"
                        y2="2.60057"
                        gradientUnits="userSpaceOnUse"
                    >
                        <stop stopColor="#08C4FF" />
                        <stop offset="0.5" stopColor="#4892FF" />
                        <stop offset="1" stopColor="#8A61FF" />
                    </linearGradient>
                </defs>
            </svg>
        </span>
    );
};
