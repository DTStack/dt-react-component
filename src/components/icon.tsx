import React, { CSSProperties } from 'react';

interface IIconProps extends React.HTMLAttributes<HTMLSpanElement> {
    title?: string;
    className?: string;
    style?: CSSProperties;
    onClick?: () => void;
}

export function SearchIcon({ title, style, className }: IIconProps) {
    return (
        <span title={title || ''} className={className} style={style}>
            <svg
                width="1em"
                height="1em"
                className="icon"
                viewBox="0 0 80 80"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path
                    d="M41.8008 68.8008C58.9216 68.8008 72.8008 54.9216 72.8008 37.8008C72.8008 20.68 58.9216 6.80078 41.8008 6.80078C24.68 6.80078 10.8008 20.68 10.8008 37.8008C10.8008 54.9216 24.68 68.8008 41.8008 68.8008Z"
                    fill="#E8F1FF"
                    stroke="#BBD6FF"
                    strokeWidth="0.8"
                />
                <path
                    fillRule="evenodd"
                    clipRule="evenodd"
                    d="M37.8635 12.8008L50.8016 28.2243V63.8539C50.8016 64.4468 50.5613 64.9835 50.1728 65.372C49.7842 65.7605 49.2475 66.0008 48.6547 66.0008H11.7484C11.1556 66.0008 10.6189 65.7605 10.2304 65.372C9.84186 64.9835 9.60156 64.4468 9.60156 63.8539V14.9477C9.60156 14.3548 9.84186 13.8181 10.2304 13.4296C10.6189 13.0411 11.1556 12.8008 11.7484 12.8008H37.8635Z"
                    fill="white"
                    stroke="#BBD6FF"
                    strokeWidth="0.8"
                />
                <path
                    d="M19.152 18.068V19.2006C19.1608 19.3622 19.0716 19.5133 18.9256 19.5845C17.7277 20.2695 16.5334 20.9591 15.3351 21.6426C15.1 21.7738 14.9906 21.962 14.9906 22.2315C14.9947 23.7338 14.9947 25.2357 14.9906 26.7373C14.9896 27.0114 15.099 27.2006 15.3381 27.3369C16.6467 28.0841 17.9531 28.8346 19.2573 29.5883C19.4939 29.7235 19.7151 29.7261 19.9538 29.5883C21.262 28.8329 22.5718 28.0806 23.8831 27.3313C24.1182 27.197 24.224 27.0084 24.224 26.7393C24.2233 26.354 24.2229 25.9105 24.2225 25.4265L24.2222 25.0021L24.2222 24.8932L24.2218 24.1038L24.2218 23.9876L24.2214 20.4547L25.4719 21.1722V27.9116L19.6073 31.2812L13.7422 27.9116V21.1722L19.152 18.068ZM16.979 27.2241L16.9714 27.2197L16.8982 27.1777L16.7426 27.0885L16.7378 27.0857L16.6542 27.0377L16.6506 27.0357L16.5606 26.9841L16.5578 26.9825L16.5268 26.9646L16.491 26.9443C16.4788 26.9372 16.4666 26.9299 16.454 26.9223L16.4279 26.9066L16.3704 26.8726C16.4057 26.8507 16.4226 26.8395 16.4394 26.8299L17.9131 25.9843L17.9686 25.9521L19.3666 25.145C19.3888 25.1321 19.4108 25.1208 19.4328 25.1112C19.4402 25.1084 19.4474 25.1053 19.4546 25.1023C19.4591 25.1006 19.4636 25.099 19.4682 25.0974L19.4734 25.0952C19.5376 25.0724 19.6005 25.0669 19.6635 25.0768C19.6679 25.0779 19.6723 25.0787 19.6767 25.0795C19.6896 25.0816 19.7023 25.0846 19.7152 25.0883C19.7188 25.0897 19.7224 25.0908 19.726 25.0919C19.7654 25.1039 19.8051 25.1219 19.8454 25.145C20.8099 25.7 21.7737 26.2547 22.7369 26.809C22.7469 26.8148 22.7569 26.8208 22.7673 26.8272L22.7721 26.8299C22.79 26.8395 22.8059 26.8507 22.8411 26.8726C22.8205 26.8848 22.8015 26.896 22.7837 26.9066L22.7577 26.9223L22.7205 26.9443C21.7669 27.4925 20.8108 28.0382 19.8587 28.5905C19.8504 28.5954 19.842 28.6001 19.8338 28.6045L19.8081 28.6175L19.7958 28.6232C19.7827 28.6292 19.7694 28.6346 19.7562 28.6394C19.7525 28.6406 19.7486 28.6419 19.7447 28.6432C19.7312 28.6479 19.7178 28.6517 19.7044 28.6549C19.5869 28.6819 19.4752 28.6599 19.3528 28.5905L17.9231 27.7659L17.1758 27.3369L17.1054 27.2965L17.049 27.2641L16.979 27.2241ZM19.1499 20.5055V20.6367C19.1499 21.7342 19.1479 22.8316 19.1499 23.9291C19.1625 24.1147 19.0617 24.2896 18.8944 24.3725C17.9373 24.9172 16.9837 25.4675 16.0291 26.0157C15.9953 26.035 15.9606 26.0513 15.9059 26.0792V25.9363C15.9059 24.8465 15.9105 23.7572 15.9059 22.6674C15.8929 22.4739 15.999 22.2919 16.1742 22.2071C17.1288 21.666 18.0783 21.1158 19.0298 20.5686C19.0636 20.5498 19.0988 20.533 19.1499 20.5055ZM20.0642 20.5075C20.1158 20.535 20.1505 20.5518 20.1843 20.5706L22.0867 21.6635L23.0399 22.2071C23.2132 22.3053 23.2985 22.4319 23.3062 22.6191V26.0747C23.2632 26.0548 23.2321 26.0431 23.204 26.0269L21.191 24.8725L21.0162 24.7707L20.287 24.3532L20.2737 24.3451C20.148 24.2672 20.0695 24.1323 20.0642 23.985V20.5075ZM20.0626 18.0645L23.3113 19.9303V21.3151C22.3049 20.7361 21.2973 20.1583 20.2885 19.582C20.1425 19.5108 20.0535 19.3596 20.0626 19.198C20.064 18.8203 20.064 18.4425 20.0626 18.0645Z"
                    fill="#BBD6FF"
                />
                <path d="M13.6719 38.3652H34.5183" stroke="#DDEBFF" strokeWidth="1.27344" />
                <path d="M13.6875 46.3652L46.5814 46.5778" stroke="#DDEBFF" strokeWidth="1.27344" />
                <path d="M13.6875 54.5781L46.5814 54.7907" stroke="#DDEBFF" strokeWidth="1.27344" />
                <path
                    fillRule="evenodd"
                    clipRule="evenodd"
                    d="M38.0508 12.4004V26.3804C38.0508 27.3182 38.811 28.0784 39.7487 28.0784H51.2023L38.0508 12.4004Z"
                    fill="#C0C3D1"
                />
                <path
                    fillRule="evenodd"
                    clipRule="evenodd"
                    d="M68.0554 19.7351L73.8314 35.6116L60.7971 71.4229C60.5943 71.98 60.185 72.4021 59.687 72.6344C59.1891 72.8666 58.6025 72.9088 58.0454 72.706L23.3649 60.0833C22.8078 59.8806 22.3856 59.4712 22.1534 58.9732C21.9212 58.4753 21.879 57.8888 22.0818 57.3317L38.8087 11.3748C39.0115 10.8177 39.4208 10.3956 39.9188 10.1634C40.4167 9.93117 41.0033 9.88893 41.5604 10.0917L68.0554 19.7351Z"
                    fill="white"
                    stroke="#BBD6FF"
                    strokeWidth="0.8"
                />
                <path d="M40.0078 21.0918L59.597 28.2217" stroke="#DDEBFF" strokeWidth="1.27344" />
                <path d="M36.9531 29.2852L62.9968 38.8758" stroke="#DDEBFF" strokeWidth="1.27344" />
                <path d="M33.5117 39.084L64.3492 50.5342" stroke="#DDEBFF" strokeWidth="1.27344" />
                <path d="M30.0508 48.5977L60.8883 60.0478" stroke="#DDEBFF" strokeWidth="1.27344" />
                <path
                    fillRule="evenodd"
                    clipRule="evenodd"
                    d="M68.3679 19.4238L63.5866 30.1518C63.2049 31.0083 63.5898 32.0121 64.4463 32.3939C64.4996 32.4176 64.5542 32.4386 64.6096 32.4568L74.2574 35.6124L68.3679 19.4238Z"
                    fill="#5D9EFF"
                />
            </svg>
        </span>
    );
}
export function LoupeIcon({ title, style, className }: IIconProps) {
    return (
        <span title={title || ''} className={className} style={style}>
            <svg
                className="icon"
                width="1em"
                height="1em"
                viewBox="0 0 38 38"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path
                    d="M16 29C23.1797 29 29 23.1797 29 16C29 8.8203 23.1797 3 16 3C8.8203 3 3 8.8203 3 16C3 23.1797 8.8203 29 16 29Z"
                    fill="white"
                    stroke="#BBD6FF"
                    strokeWidth="3.2"
                />
                <path
                    d="M16.4048 8.60156C14.9672 8.60156 13.8405 9.01695 13.0246 9.8675C12.2087 10.6983 11.8008 11.8257 11.8008 13.2697H13.8405C13.8405 12.3994 14.0154 11.7268 14.365 11.2521C14.7535 10.6785 15.3946 10.4016 16.2688 10.4016C16.9681 10.4016 17.5315 10.5994 17.92 10.995C18.2891 11.3906 18.4833 11.9246 18.4833 12.6169C18.4833 13.1312 18.2891 13.606 17.9394 14.0609L17.6092 14.4367C16.4048 15.5246 15.6666 16.3356 15.414 16.8895C15.1421 17.4038 15.0255 18.0367 15.0255 18.7686V19.1049H17.1041V18.7686C17.1041 18.2939 17.2012 17.8587 17.4149 17.4631C17.5897 17.1071 17.8617 16.751 18.2308 16.4345C19.1438 15.6235 19.6878 15.1093 19.8626 14.8917C20.3482 14.2587 20.6008 13.4477 20.6008 12.4785C20.6008 11.2917 20.2123 10.3422 19.4352 9.64991C18.6582 8.93783 17.648 8.60156 16.4048 8.60156ZM16.0551 20.173C15.6666 20.173 15.3169 20.2917 15.0449 20.5686C14.773 20.8257 14.637 21.162 14.637 21.5774C14.637 21.973 14.773 22.3093 15.0449 22.5862C15.3169 22.8631 15.6666 23.0016 16.0551 23.0016C16.4436 23.0016 16.7933 22.8631 17.0652 22.606C17.3372 22.329 17.4926 21.9928 17.4926 21.5774C17.4926 21.162 17.3566 20.8257 17.0847 20.5686C16.8127 20.2917 16.463 20.173 16.0551 20.173Z"
                    fill="#5D9EFF"
                />
                <path
                    d="M26 25.6016L34.8 34.4016"
                    stroke="#BBD6FF"
                    strokeWidth="4"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                />
            </svg>
        </span>
    );
}
