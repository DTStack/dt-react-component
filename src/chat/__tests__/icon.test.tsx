import React from 'react';
import { cleanup, render } from '@testing-library/react';

import { GradientDotIcon, SendColored } from '../icon';

describe('Test Chat Icon', () => {
    beforeEach(cleanup);

    it('Match Snapshots', () => {
        expect(render(<SendColored />).asFragment()).toMatchSnapshot('SendIcon with gradient');
        expect(render(<GradientDotIcon />).asFragment()).toMatchSnapshot('GradientDotIcon');
    });
});
