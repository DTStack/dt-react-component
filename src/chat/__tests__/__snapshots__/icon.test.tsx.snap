// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Chat Icon Match Snapshots: GradientDotIcon 1`] = `
<DocumentFragment>
  <span
    class="dtc__icon"
  >
    <svg
      fill="none"
      height="1em"
      viewBox="0 0 6 6"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 3C0 1.34315 1.34315 0 3 0C4.65685 0 6 1.34315 6 3C6 4.65685 4.65685 6 3 6C1.34315 6 0 4.65685 0 3Z"
        fill="url(#paint0_linear_5253_12831)"
      />
      <circle
        cx="3"
        cy="3"
        fill="url(#paint1_linear_5253_12831)"
        r="3"
      />
      <defs>
        <lineargradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear_5253_12831"
          x1="0"
          x2="7.2238"
          y1="0"
          y2="2.60057"
        >
          <stop
            stop-color="#08C4FF"
          />
          <stop
            offset="0.5"
            stop-color="#4892FF"
          />
          <stop
            offset="1"
            stop-color="#8A61FF"
          />
        </lineargradient>
        <lineargradient
          gradientUnits="userSpaceOnUse"
          id="paint1_linear_5253_12831"
          x1="0"
          x2="7.2238"
          y1="0"
          y2="2.60057"
        >
          <stop
            stop-color="#08C4FF"
          />
          <stop
            offset="0.5"
            stop-color="#4892FF"
          />
          <stop
            offset="1"
            stop-color="#8A61FF"
          />
        </lineargradient>
      </defs>
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Test Chat Icon Match Snapshots: SendIcon with gradient 1`] = `
<DocumentFragment>
  <span
    class="dtc__icon"
  >
    <svg
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M17.999 3.858 3.35 9.204a1.66 1.66 0 0 0-1.095 1.458 1.66 1.66 0 0 0 .862 1.602l4.454 2.44c.27.143.593.098.808-.118l.737-.72 4.508-4.428a.677.677 0 0 1 .952.018c.26.261.26.693-.009.954l-4.562 4.473-.61.603a.68.68 0 0 0-.118.828l2.64 4.626c.315.513.845.81 1.429.81.063 0 .117 0 .18-.009a1.68 1.68 0 0 0 1.41-1.107L20.153 6a1.69 1.69 0 0 0-.404-1.764 1.64 1.64 0 0 0-1.751-.378"
        fill="url(#paint0_linear_3878_6538)"
        fill-rule="evenodd"
      />
      <defs>
        <lineargradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear_3878_6538"
          x1="2.25"
          x2="23.921"
          y1="3.75"
          y2="11.552"
        >
          <stop
            stop-color="#00BAC6"
          />
          <stop
            offset="0.5"
            stop-color="#0067FF"
          />
          <stop
            offset="1"
            stop-color="#450FDE"
          />
        </lineargradient>
      </defs>
      <defs>
        <lineargradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear_3878_6538_hover"
          x1="2.25"
          x2="23.921"
          y1="3.75"
          y2="11.552"
        >
          <stop
            stop-color="#08C4FF"
          />
          <stop
            offset="0.5"
            stop-color="#4892FF"
          />
          <stop
            offset="1"
            stop-color="#8A61FF"
          />
        </lineargradient>
      </defs>
    </svg>
  </span>
</DocumentFragment>
`;
